src/Aspire.Hosting.PostgreSQL/PostgresBuilderExtensions.cs

```diff
@@ -162,39 +161,21 @@ public static IResourceBuilder<T> WithPgAdmin<T>(this IResourceBuilder<T> builde
 
             builder.ApplicationBuilder.Eventing.Subscribe<AfterEndpointsAllocatedEvent>((e, ct) =>
             {
-                // Add the servers.json file bind mount to the pgAdmin container
-
                 var postgresInstances = builder.ApplicationBuilder.Resources.OfType<PostgresServerResource>();
 
-                // Create servers.json file content in a temporary file
-
-                var tempConfigFile = WritePgAdminServerJson(postgresInstances);
-
-                try
-                {
-                    var aspireStore = e.Services.GetRequiredService<IAspireStore>();
-
-                    // Deterministic file path for the configuration file based on its content
-                    var configJsonPath = aspireStore.GetFileNameWithContent($"{builder.Resource.Name}-servers.json", tempConfigFile);
-
-                    // Need to grant read access to the config file on unix like systems.
-                    if (!OperatingSystem.IsWindows())
-                    {
-                        File.SetUnixFileMode(configJsonPath, FileMode644);
-                    }
+                // Generate the contents of the servers.json file
+                var serversConfigFile = WritePgAdminServerJson(postgresInstances);
 
-                    pgAdminContainerBuilder.WithBindMount(configJsonPath, "/pgadmin4/servers.json");
-                }
-                finally
-                {
-                    try
+                // Create a servers.json file in the container with necessary permissions
+                pgAdminContainerBuilder.WithCreateFile("/pgadmin4", new()
```

Comment:
`WithCreateFile` seems like an odd name for this API given what it is really doing is `docker cp`.

Reply:
I would say that `ContainerCreateFileAnnotation` is an implementation detail and here we care about providing a custom file.

But we could also say that `docker cp` is an implementation detail and from a user's perspective we want to create a file in the container.

Personally, I may go with `WithContainerFile(name, callback)` this way it hides both implementation details ;)

Reply:
I like `WithContainerFile(s)`.

Reply:
Me too!